src/base/Permit2Forwarder.sol

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.24;
+
+import {IAllowanceTransfer} from "permit2/src/interfaces/IAllowanceTransfer.sol";
+
+/// @notice PermitForwarder allows permitting this contract as a spender on permit2
+/// @dev This contract does not enforce the spender to be this contract, but that is the intended use case
+/// TODO: Does it need to be enforced?
+contract Permit2Forwarder {
+    IAllowanceTransfer public immutable permit2;
+
+    constructor(IAllowanceTransfer _permit2) {
+        permit2 = _permit2;
+    }
+
+    /// @notice allows forwarding a single permit to permit2
+    function selfPermit(address owner, IAllowanceTransfer.PermitSingle calldata permitSingle, bytes calldata signature)
```

Comment:
This is just to allow it to be added into multicall right? but in reality they could be permitting anyone, it doesnt have to be `self`?

If so I'd propose just calling it `permit`? `Permit2Forwarder.permit` makes sense to me

or `permitSingle` and `permitBatch`?

Reply:
oh just seen your `@dev` above!
No i dont think it needs to be enforced, I think just rename to remove the word `self` and keep the comment that thats the intended use case